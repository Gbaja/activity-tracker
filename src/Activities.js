import React, {Component} from "react";
import {Link} from "react-router-dom"

import Header from "./Header";
import firebase from "./firebase.js";

class Activities extends Component{
    constructor() {
        super();
        this.state = { items: [] };
    }
    componentDidMount() {
        const itemsRef = firebase.database().ref('items');
        itemsRef.on('value', (snapshot) => {
            let items = snapshot.val();
            let newState = [];
            for (let item in items) {
                newState.push({
                    id: item,
                    date: items[item].date,
                    notes: items[item].notes
                });
            }
            this.setState({
                items: newState
            });
        });
    }
    removeItem(itemId) {
        const itemRef = firebase.database().ref(`/items/${itemId}`);
        itemRef.remove();
    }
    render(){
       return <React.Fragment>
           <div className="app">
             <Header />
             <section className="display-item">
               <div>
                 <button>
                   <Link className="anchor_tags" to="/">
                     Back
                   </Link>
                 </button>
               </div>
               <div className="wrapper">
                 <ul>
                   {this.state.items.map(item => {
                     return <li key={item.id}>
                         <h3>Date: {item.date}</h3>
                         <p>Notes: {item.notes}</p>
                         <button
                           onClick={() => this.removeItem(item.id)}
                         >
                           Remove Item
                         </button>
                       </li>;
                   })}
                 </ul>
               </div>
             </section>
           </div>
         </React.Fragment>;
    }
}

export default Activities;