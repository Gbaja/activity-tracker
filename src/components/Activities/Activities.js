import React, {Fragment} from "react";
import {Link} from "react-router-dom";

import Header from "../Header"

const Activities = props => {

  const {activities, removeItem, loading} = props;

  return (
    <Fragment>
      <div className="app">
        <Header />
        <section className="display-item">
          <div>
            <button>
              <Link className="anchor_tags" to="/">
                Back
              </Link>
            </button>
          </div>
          <div className="wrapper">
            {loading ? <div> Loading </div> : <ul>
              {activities.map(activity => {
                return <li key={activity.id}>
                  <h3>Date: {activity.date}</h3>
                  <p>Notes: {activity.notes}</p>
                  <button
                    onClick={() => removeItem(activity.id)}
                  >
                    Remove Item
                          </button>
                </li>
              })}
            </ul>
            }
          </div>
        </section>
      </div>
    </Fragment>
  )
}

export default Activities;