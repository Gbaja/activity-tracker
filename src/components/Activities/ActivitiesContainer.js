import React, { Component } from "react";

import Activities from "./Activities";
import firebase from "../../firebase";

class ActivitiesContainer extends Component {
    constructor() {
        super();
        this.state = { 
            loading: false,
            items: []
        };
    }

    componentDidMount() {
        const itemsRef = firebase.database().ref("items");
        this.setState({loading: true}, ()=>{
            itemsRef.on('value', (snapshot) => {
                let items = snapshot.val();
                let newState = [];
                for (let item in items) {
                    newState.push({
                        id: item,
                        date: items[item].date,
                        notes: items[item].notes
                    });
                }
                this.setState({
                    items: newState,
                    loading: false
                });
            });
        });
        
    }

    removeItem(itemId) {
        const itemRef = firebase.database().ref(`/items/${itemId}`);
        itemRef.remove();
    }

    changeIt = (array) => {
        return array.sort((a, b) => {
            return new Date(b.date) - new Date(a.date);

        })
    }
    
    render() {
        return (
            <Activities 
            activities={this.changeIt(this.state.items)} 
            removeItem={this.removeItem} 
            loading={this.state.loading}
            />
        )
    }
}

export default ActivitiesContainer;