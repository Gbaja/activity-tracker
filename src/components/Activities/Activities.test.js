import React from "react"
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import { shallow } from "enzyme";

import Activities from "./Activities"
import ActivitiesContainer from "./ActivitiesContainer"

describe("Activities tests", ()=>{
    const activities = [
        { id: "1", date: "12/03/2018", notes: "Write tests" },
        { id: "2", date: "22/04/2018", notes: "Write more tests" },
        { id: "3", date: "02/05/2018", notes: "Write even more tests" },
    ];
    const removeItem = (id) => {
        console.log(id)
    };
    const loading = false;
    it("Activities snapshot passes", ()=>{
        const tree =  renderer.create(
            <MemoryRouter>
                <Activities 
                    activities={activities} 
                    removeItem={removeItem} 
                    loading={loading}
                />
            </MemoryRouter>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})

describe("Activities container tests", ()=>{
    it("", ()=>{
        const wrapper = shallow(
                <ActivitiesContainer />,
                {
                disableLifecycleMethods: true
        })

        expect(wrapper.state().loading).toBeFalsy();
        expect(wrapper.state().items.length).toEqual(0)
        wrapper.instance().componentDidMount();
        expect(wrapper.state().loading).toBeTruthy();
        //not passing = expect(wrapper.state().items.length).toBeGreaterThan(0)
    })
})
