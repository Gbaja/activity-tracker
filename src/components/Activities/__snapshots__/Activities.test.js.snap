// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activities tests Activities snapshot passes 1`] = `
<div
  className="app"
>
  <header>
    <div
      className="wrapper"
    >
      <h1>
        Laide's Activity tracker
      </h1>
    </div>
  </header>
  <section
    className="display-item"
  >
    <div>
      <button>
        <a
          className="anchor_tags"
          href="/"
          onClick={[Function]}
        >
          Back
        </a>
      </button>
    </div>
    <div
      className="wrapper"
    >
      <ul>
        <li>
          <h3>
            Date: 
            12/03/2018
          </h3>
          <p>
            Notes: 
            Write tests
          </p>
          <button
            onClick={[Function]}
          >
            Remove Item
          </button>
        </li>
        <li>
          <h3>
            Date: 
            22/04/2018
          </h3>
          <p>
            Notes: 
            Write more tests
          </p>
          <button
            onClick={[Function]}
          >
            Remove Item
          </button>
        </li>
        <li>
          <h3>
            Date: 
            02/05/2018
          </h3>
          <p>
            Notes: 
            Write even more tests
          </p>
          <button
            onClick={[Function]}
          >
            Remove Item
          </button>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
